### ISSUE TITLE
Roslyn ignores [MemberNotNull] annotation when checking properties initialization in constructors

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#44180]
**Version Used**: 

```
commit 7b27ca3b7722c5e56910a6cfba837e6a9d352d4f
Author: Tom<C3><A1><C5><A1> Matou<C5><A1>ek <tmat@users.noreply.github.com>
Date:   Mon May 11 19:17:04 2020 -0700

    Fixes PathMap parsing to support all paths (#43706)
```

**Steps to Reproduce**:

Compile the following code
```cs
using System.Diagnostics.CodeAnalysis;

class Property
{
  public string P { get; set; }

  public Property() { Init(); P.ToString(); } // false CS8618: Non-nullable property 'P' is uninitialized. Consider declaring the property as nullable.

  [MemberNotNull(nameof(P))]  void Init() => P = "";
}

class Field
{
  public string F;
  
  public Field() { Init(); F.ToString(); }

  [MemberNotNull(nameof(F))] void Init() => F = "";
}
```

**Expected Behavior**:
No warnings. Both classes calls members that guarantee to initialize their members in constructors and nullable inspector recognizes it and doesn't issue warnings for their dereferences.

**Actual Behavior**:
`CS8618: Non-nullable property 'P' is uninitialized. Consider declaring the property as nullable.` is reported for the constructor in class `Property`
